src/test/java/com/sharewalk/controller/WalkControllerTest.java

```diff
@@ -0,0 +1,142 @@
+package com.sharewalk.controller;
+
+import com.sharewalk.dao.WalkDAO;
+import com.sharewalk.model.User;
+import com.sharewalk.model.Walk;
+import com.sharewalk.service.WalkService;
+import junitparams.JUnitParamsRunner;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+@RunWith(JUnitParamsRunner.class)
+public class WalkControllerTest {
+    @Rule
+    public MockitoRule mockito = MockitoJUnit.rule();
+
+    @Rule
+    public ExpectedException ex = ExpectedException.none();
+
+    @InjectMocks
+    private WalkController instance;
+
+    @Mock
+    private WalkDAO walkDAO;
+
+    @Test
+    public void getAllWalksGeneralTest(){
+        when(walkDAO.listWalks()).thenReturn(Arrays.asList(new Walk("walk 1", null, null), new Walk("walk 2", null, null) ));
+        ResponseEntity responseEntity = instance.getAllWalks(null);
+        assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
+        assertEquals(2, ((List)responseEntity.getBody()).size());
+        assertEquals(Walk.class, ((List)responseEntity.getBody()).get(0).getClass());
+    }
+
+    @Test
+    public void getAllWalksStartsWithTestOK(){
+        when(walkDAO.listWalks("walk 1")).thenReturn(Arrays.asList(new Walk("walk 1", null, null)));
```

Comment:
you don't need this.
Check only if dao method was called with proper argument, and if the result of dao was returned by the controller (can be empty list)